Python/sysmodule.c

```diff
@@ -2521,6 +2517,10 @@ PyAPI_FUNC(int) PyUnstable_CopyPerfMapFile(const char* parent_filename) {
     }
     char buf[4096];
     PyThread_acquire_lock(perf_map_state.map_lock, 1);
+    FILE* from = fopen(parent_filename, "r");
+    if (!from) {
```

Comment:
You forgot to release lock here.

Reply:
moved fopen to before PyThread_acquire_lock to satisfy both PR conversations.